### PR TITLE
drivers/kernel_test: rename the kernel test driver

### PR DESCRIPTION
**Pull request by deepaksrma**
- rename the kernel_test_drv as os_test_drv because kernel_test contains kernel and network TCs and also may contain filesystem TCs.
- remove unnecessary header files

Signed-off-by: Deepak sharma <deepak.sh@samsung.com>

### CODE REVIEW

**Comment by sunghan-chang** [comment]
@deepaksrma Please find coding rule violations.

**Comment by deepaksrma** [comment]
> @deepaksrma Please find coding rule violations.

corrected

File: apps/examples/mpu/mpu_test.c [hunk 1]
```diff
@@ -38,9 +38,9 @@ int mpu_tc_main(int argc, char *argv[])
 	struct mputest_arg_s obj;
 	int ret;
 
-	tc_fd = open(KERNEL_TEST_DRVPATH, O_WRONLY);
+	tc_fd = open(OS_API_TEST_DRVPATH, O_WRONLY);
 	if (tc_fd < 0) {
-		printf("\nFAILED to open kernel test driver %s\n", KERNEL_TEST_DRVPATH);
+		printf("\nFAILED to open kernel test driver %s\n", OS_API_TEST_DRVPATH);
```
Comment (sangam-swami) [inline apps/examples/mpu/mpu_test.c:43]: Failed to open os api test driver

File: apps/examples/rdp/rdp_main.c [hunk 2]
```diff
@@ -83,10 +83,10 @@ int rdp_main(int argc, char *argv[])
 	int tc_fd = 0;
 
 	printf("rdp main!!\n");
-	tc_fd = open(KERNEL_TEST_DRVPATH, O_WRONLY);
+	tc_fd = open(OS_API_TEST_DRVPATH, O_WRONLY);
 
 	if (tc_fd < 0) {
-		printf("FAILED to open kernel test driver %s\n", KERNEL_TEST_DRVPATH);
+		printf("FAILED to open kernel test driver %s\n", OS_API_TEST_DRVPATH);
```
Comment (sangam-swami) [inline apps/examples/rdp/rdp_main.c:89]: same here too

File: apps/examples/testcase/le_tc/network/tc_net_pbuf.c [hunk 3]
```diff
@@ -39,7 +39,7 @@
 * @brief		:check for the memory allocation for the pbuf by transport layer
 * @scenario		:
 * @apicovered		:pbuf_alloc()
-* @precondition 	:
+* @precondition		:
```
Comment (sangam-swami) [inline apps/examples/testcase/le_tc/network/tc_net_pbuf.c:42]: why extra tab added

File: os/include/tinyara/fs/ioctl.h [hunk 4]
```diff
@@ -161,10 +161,9 @@
 										 * OUT: Bytes writable to this fd
 										 */
 
-#define FIONBIO         _FIOC(0x000b)     /* IN:  Boolean option takes an
-                                           *      int value.
-                                           * OUT: Origin option.
-                                           */
+#define FIONBIO         _FIOC(0x000b)     /* IN:  Boolean option takes an int value.
+										 * OUT: Origin option.
+										 */
```
Comment (sangam-swami) [inline os/include/tinyara/fs/ioctl.h:166]: why these changes needed

File: os/include/tinyara/os_api_test_drv.h [hunk 5]
```diff
@@ -103,21 +103,21 @@ extern "C" {
...
-void kernel_test_drv_register(void);
+void os_api_test_drv_register(void);
 
 #undef EXTERN
 #ifdef __cplusplus
 }
 #endif
 
-#endif							/* CONFIG_DRIVERS_KERNEL_TEST */
+#endif							/* CONFIG_DRIVERS_OS_API_TEST */
```
Comment (sangam-swami) [inline os/include/tinyara/os_api_test_drv.h:122]: __INCLUDE_TINYARA_OS_API_TEST_DRV_H for below #endif comment

**Comment by sangam-swami** [comment]
is the PR changes tested after executing tc's?

**Comment by sunghan-chang** [comment]
@deepaksrma Could you check comments above?

File: apps/examples/testcase/le_tc/network/tc_net_pbuf.c (hunk 3 above)
Comment (deepaksrma) [reply apps/examples/testcase/le_tc/network/tc_net_pbuf.c:42]: because earlier, after * @precondition there was a space then tab, so after raising PR it gave me coding rules violations error.
so this is why i had to use extra tab but at vim editor these are aligned. not showing like here.

File: os/include/tinyara/fs/ioctl.h (hunk 4 above)
Comment (deepaksrma) [reply os/include/tinyara/fs/ioctl.h:166]: because of coding rules violation error

**Comment by deepaksrma** [comment]
> is the PR changes tested after executing tc's?

yes. PR got tested for rtl8721csm/tc. and all TCs got passed.

File: os/include/tinyara/os_api_test_drv.h (hunk 5 above)
Comment (deepaksrma) [reply os/include/tinyara/os_api_test_drv.h:122]: corrected

File: apps/examples/rdp/rdp_main.c (hunk 2 above)
Comment (deepaksrma) [reply apps/examples/rdp/rdp_main.c]: corrected

File: apps/examples/mpu/mpu_test.c (hunk 1 above)
Comment (deepaksrma) [reply apps/examples/mpu/mpu_test.c]: corrected

File: apps/examples/testcase/le_tc/network/tc_net_pbuf.c (hunk 3 above)
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/tc_net_pbuf.c:42]: I think removing a space is right. This looks it is not aligned.
(You gave a reply to unknown ID)

File: apps/examples/testcase/le_tc/network/tc_net_pbuf.c [hunk 6]
```diff
@@ -118,7 +118,7 @@ static void tc_net_pbuf_layer_raw_tx_p(int fd)
 	args.type = PBUF_RAM;
 
 	ret = ioctl(fd, TESTIOC_NET_PBUF, &args);
-	TC_ASSERT_GEQ("pbuf",ret, 1);
+	TC_ASSERT_GEQ("pbuf", ret, 1);
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/tc_net_pbuf.c:121]: Please split a commit for fixing coding rule violations.

File: build/configs/stm32l4r9ai-disco/hello/defconfig [hunk 7]
```diff
@@ -682,7 +682,7 @@ CONFIG_FB_OVERLAY=y
 # CONFIG_WATCHDOG is not set
 # CONFIG_TIMER is not set
 # CONFIG_ANALOG is not set
-# CONFIG_DRIVERS_KERNEL_TEST is not set
+# CONFIG_DRIVERS_OS_API_TEST is not set
```
Comment (sunghan-chang) [inline build/configs/stm32l4r9ai-disco/hello/defconfig:685]: We prefer splitting a commit for this. Because this makes difficult to see real change of this commit.

File: os/drivers/os_api_test/network/test_net_pbuf.c [hunk 8]
```diff
@@ -56,7 +56,7 @@ int test_net_pbuf(int cmd, unsigned long arg)
 {
 	int ret = -1;
 
-	switch(cmd) {
+	switch (cmd) {
```
Comment (sunghan-chang) [inline os/drivers/os_api_test/network/test_net_pbuf.c:59]: same as previous

File: os/include/tinyara/fs/ioctl.h (hunk 4 above)
Comment (sunghan-chang) [reply os/include/tinyara/fs/ioctl.h:166]: same as previous
but for this case, if you use only spaces in this line, then no violation for indentation.
I prefer spaces because of current view. (Github shows not aligned indendation with tabs)

File: apps/examples/testcase/le_tc/network/tc_net_pbuf.c (hunk 3 above)
Comment (deepaksrma) [reply apps/examples/testcase/le_tc/network/tc_net_pbuf.c:42]: yes. here it is looking not aligned but in linux machine(vim editor), it looks aligned.
now you suggest what to do?

Comment (sunghan-chang) [reply apps/examples/testcase/le_tc/network/tc_net_pbuf.c:42]: I think you changed a space to a tab. How about just removing a space?

Comment (deepaksrma) [reply apps/examples/testcase/le_tc/network/tc_net_pbuf.c:42]: if i remove a space then at vi editor, it shows me like this
* @precondition :

File: apps/examples/testcase/le_tc/network/tc_net_pbuf.c (hunk 6 above)
Comment (deepaksrma) [reply apps/examples/testcase/le_tc/network/tc_net_pbuf.c:121]: ok.

File: os/include/tinyara/fs/ioctl.h (hunk 4 above)
Comment (deepaksrma) [reply os/include/tinyara/fs/ioctl.h:166]: yes. but it gave me coding rules violation error while i hadn't touched these files.
so to fix these errors. i made these changes

File: apps/examples/testcase/le_tc/network/tc_net_pbuf.c (hunk 3 above)
Comment (sunghan-chang) [reply apps/examples/testcase/le_tc/network/tc_net_pbuf.c:42]: You should consider Github syntax.... `@precondtion` which you wrote is in c code. But without any other syntax, `@precondition` sends a mail to `precondition` Github ID. He already got two mail because of this.

Comment (sunghan-chang) [reply apps/examples/testcase/le_tc/network/tc_net_pbuf.c:42]: @deepaksrma You must take care of using `@` in Github comment.

Comment (deepaksrma) [reply apps/examples/testcase/le_tc/network/tc_net_pbuf.c:42]: ok

File: os/drivers/os_api_test/Make.defs [hunk 9]
```diff
@@ -16,10 +16,10 @@
 #
 ###########################################################################
 
-ifeq ($(CONFIG_DRIVERS_KERNEL_TEST),y)
```
Comment (kishore-sn) [inline os/drivers/os_api_test/Make.defs:19]: This config has not been renamed in  os/drivers/os_api_test/Kconfig file.

File: build/configs/artik053/grpc/defconfig [hunk 10]
```diff
@@ -425,7 +425,7 @@ CONFIG_ADC=y
 CONFIG_ADC_FIFOSIZE=8
 CONFIG_ADC_NPOLLWAITERS=2
 # CONFIG_DAC is not set
-# CONFIG_DRIVERS_KERNEL_TEST is not set
+# CONFIG_DRIVERS_OS_API_TEST is not set
```
Comment (kishore-sn) [inline build/configs/artik053/grpc/defconfig:428]: This config is used in many Kconfig files. 
Please check and change in all these places. 

apps/examples/testcase/le_tc/drivers/Kconfig:	select DRIVERS_KERNEL_TEST
apps/examples/testcase/le_tc/compression/Kconfig:	select DRIVERS_KERNEL_TEST
apps/examples/testcase/le_tc/network/Kconfig:	select DRIVERS_KERNEL_TEST
apps/examples/testcase/le_tc/kernel/Kconfig:	select DRIVERS_KERNEL_TEST
apps/examples/rdp/Kconfig:	select DRIVERS_KERNEL_TEST
apps/examples/mpu/Kconfig:	select DRIVERS_KERNEL_TEST
apps/examples/stack_protection/Kconfig:	select DRIVERS_KERNEL_TEST

File: os/drivers/Kconfig [hunk 11]
```diff
@@ -431,7 +431,7 @@ if ANALOG
 source drivers/analog/Kconfig
 endif # ANALOG
 
-source drivers/kernel_test/Kconfig
+source drivers/os_api_test/Kconfig
```
Comment (kishore-sn) [inline os/drivers/Kconfig:434]: Please check the naming in below Kconfig also.

apps/examples/testcase/le_tc/kernel/Kconfig:	bool "Kernel TestCase Example"
apps/examples/testcase/le_tc/kernel/Kconfig:		Enable the Kernel TestCase Example
apps/examples/testcase/le_tc/kernel/Kconfig:	bool "Kernel Testcase Debug Output"

File: os/drivers/os_api_test/kernel/Make.defs [hunk 12]
```diff
@@ -46,7 +46,7 @@ CFLAGS += -I$(TOPDIR)/kernel
 
 # Include kernel test driver support
```
Comment (kishore-sn) [inline os/drivers/os_api_test/kernel/Make.defs:47]: kernel test -> os api test

File: os/drivers/os_api_test/kernel/Make.defs [hunk 13]
```diff
@@ -17,7 +17,7 @@
 ############################################################################
 # Include kernel test drivers
```
Comment (kishore-sn) [inline os/drivers/os_api_test/kernel/Make.defs:18]: kernel test -> os api test

File: os/drivers/os_api_test/kernel/Make.defs [hunk 14]
```diff
@@ -17,7 +17,7 @@
...
 
-ifeq ($(CONFIG_DRIVERS_KERNEL_TEST),y)
```
Comment (kishore-sn) [inline os/drivers/os_api_test/kernel/Make.defs:20]: Also, pls search for "kernel test" in whole code and where ever required, pls change it to "os api test"

File: apps/examples/testcase/le_tc/network/tc_net_pbuf.c (hunk 3 above)
Comment (precondition) [reply apps/examples/testcase/le_tc/network/tc_net_pbuf.c:42]: Don't worry about me

File: os/include/tinyara/os_api_test_drv.h [hunk 15]
```diff
@@ -30,18 +30,18 @@
...
  ****************************************************************************/
 /* Configuration
  *
- * CONFIG_DRIVERS_KERNEL_TEST - Enables kernel_test driver support
+ * CONFIG_DRIVERS_OS_API_TEST - Enables os_api_test driver support
  */
 
 /* IOCTL Commands */
-/* The kernel_test_drv module uses ioctl commands to identify the kernel module for which
+/* The os_api_test_drv module uses ioctl commands to identify the kernel module for which
```
Comment (jeongchanKim) [inline os/include/tinyara/os_api_test_drv.h:44]: Please change the description also, because, as you said, this is used not only for kernel module but also for network and fs.

File: os/include/tinyara/fs/ioctl.h [hunk 16]
```diff
@@ -394,7 +393,7 @@
 #define _IOTBUSIOC(nr)       _IOC(_IOTBUSBASE, nr)
 
 /* Kernel_tc driver ioctl definitions *************************************/
```
Comment (jeongchanKim) [inline os/include/tinyara/fs/ioctl.h:395]: Please modify this comment also.

File: apps/examples/testcase/le_tc/network/tc_net_pbuf.c (hunk 3 above)
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/tc_net_pbuf.c:42]: 1. As I left before, please split commit for formatting. You should not include both renaming and formatting.
2. Because of space and tab mixing, it shows coding rule violation. And because of using tab, it shows different alignment at each system. That's why I prefer using spaces instead of tabs.

**Comment by sunghan-chang** [comment]
I asked you split commits as below.
1. renaming
2. modifying in defconfig
3. formatting for coding rule

File: apps/examples/testcase/le_tc/network/tc_net_pbuf.c (hunk 3 above)
Comment (deepaksrma) [reply apps/examples/testcase/le_tc/network/tc_net_pbuf.c:42]: yes. i'll split the commit.

**Comment by sunghan-chang** [comment]
@deepaksrma Any update?

**Comment by deepaksrma** [comment]
> @deepaksrma Any update?

done. please check.

File: apps/examples/testcase/le_tc/kernel/Kconfig [hunk 17]
```diff
@@ -4,15 +4,15 @@
 #
 
 menuconfig EXAMPLES_TESTCASE_KERNEL
-	bool "Kernel TestCase Example"
+	bool "kernel TestCase Example"
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/Kconfig:7]: Could you let me know why you modify only `kernel`? There are many upper characters. I can't get your point for that.

File: os/include/tinyara/os_api_test_drv.h [hunk 18]
```diff
@@ -30,18 +30,18 @@
...
-#ifdef CONFIG_DRIVERS_KERNEL_TEST
+#ifdef CONFIG_DRIVERS_OS_API_TEST
 
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
 /* Configuration
  *
- * CONFIG_DRIVERS_KERNEL_TEST - Enables kernel_test driver support
+ * CONFIG_DRIVERS_OS_API_TEST - Enables kernel_test driver support
```
Comment (sunghan-chang) [inline os/include/tinyara/os_api_test_drv.h:40]: Enables OS API test driver support

File: os/include/tinyara/os_api_test_drv.h [hunk 19]
```diff
@@ -30,18 +30,18 @@
...
  */
 
 /* IOCTL Commands */
-/* The kernel_test_drv module uses ioctl commands to identify the kernel module for which
+/* The os_api_test_drv module uses ioctl commands to identify the os api module for which
```
Comment (sunghan-chang) [inline os/include/tinyara/os_api_test_drv.h:44]: second module looks unnecessary
```
The OS API test module uses ioctl commands to identify which test cases are to be run.
```

File: os/include/tinyara/os_api_test_drv.h [hunk 20]
```diff
@@ -103,21 +103,21 @@ extern "C" {
...
 
 /****************************************************************************
- * Name: kernel_test_drv_register
+ * Name: os_api_test_drv_register
  *
  * Description:
- *   This function creates a device node like "/dev/kernel_test" which will be used
- *   by the tests that execute kernel side APIs
+ *   This function creates a device node like "/dev/os_api_test" which will be used
+ *   by the tests that execute os(kernel, network and fs) side APIs
```
Comment (sunghan-chang) [inline os/include/tinyara/os_api_test_drv.h:110]: Let's write OS instead of os because it's description, not name.

File: os/include/tinyara/os_api_test_drv.h [hunk 21]
```diff
@@ -16,11 +16,11 @@
...
  ****************************************************************************/
 
-#ifndef __INCLUDE_TINYARA_KERNEL_TEST_DRV_H
-#define __INCLUDE_TINYARA_KERNEL_TEST_DRV_H
+#ifndef __INCLUDE_TINYARA_OS_API_TEST_DRV_H
+#define __INCLUDE_TINYARA_OS_API_TEST_DRV_H
 
 /* This file will be used to provide definitions to support
- * kernel test case framework
+ * os api test case framework
```
Comment (sunghan-chang) [inline os/include/tinyara/os_api_test_drv.h:23]: OS API

File: apps/examples/testcase/le_tc/kernel/Kconfig (hunk 17 above)
Comment (deepaksrma) [reply apps/examples/testcase/le_tc/kernel/Kconfig]: it was by mistake.

File: os/include/tinyara/os_api_test_drv.h (hunk 20 above)
Comment (deepaksrma) [reply os/include/tinyara/os_api_test_drv.h]: corrected

File: os/include/tinyara/os_api_test_drv.h (hunk 21 above)
Comment (deepaksrma) [reply os/include/tinyara/os_api_test_drv.h]: corrected

File: os/include/tinyara/os_api_test_drv.h (hunk 19 above)
Comment (deepaksrma) [reply os/include/tinyara/os_api_test_drv.h]: corrected

File: os/include/tinyara/os_api_test_drv.h (hunk 18 above)
Comment (deepaksrma) [reply os/include/tinyara/os_api_test_drv.h]: done

File: build/configs/artik053/STDK/defconfig [hunk 22]
```diff
@@ -436,7 +436,7 @@ CONFIG_ADC=y
 CONFIG_ADC_FIFOSIZE=8
 CONFIG_ADC_NPOLLWAITERS=2
 # CONFIG_DAC is not set
-# CONFIG_DRIVERS_KERNEL_TEST is not set
+# CONFIG_DRIVERS_OS_API_TEST is not set
```
[the remark this answers is not in the document]
Comment (deepaksrma) [reply build/configs/artik053/STDK/defconfig:439]: done

**Review by pillip8282 — COMMENTED** [review]
I've never heard os api. is it commonly used word or concept?

**Comment by deepaksrma** [comment]
> I've never heard os api. is it commonly used word or concept?

od func is suggested by sunghan, but kishore & sangam suggested os api looks better. so i changed the same

**Review by pillip8282 — APPROVED** [review]
lgtm
